x-pack/plugins/endpoint/server/routes/resolver/utils/pagination.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IScopedClusterClient } from 'kibana/server';
+import { SearchResponse } from 'elasticsearch';
+import { ResolverEvent } from '../../../../common/types';
+import { extractEventID } from './normalize';
+import { JsonObject } from '../../../../../../../src/plugins/kibana_utils/public';
+
+export interface PaginationParams {
+  size: number;
+  timestamp?: Date;
+  eventID?: string;
+}
+
+export async function getPaginationParams(
+  client: IScopedClusterClient,
+  limit: number,
+  after?: string
+): Promise<PaginationParams> {
+  if (!after) {
+    return { size: limit };
+  }
+  const cursor = (await client.callAsCurrentUser('search', {
+    body: {
+      query: {
+        ids: {
+          values: [after],
+        },
+      },
+    },
+  })) as SearchResponse<ResolverEvent>;
+
+  if (cursor.hits.hits.length === 0) {
+    return { size: limit };
+  }
+  const event = cursor.hits.hits[0]._source;
+
+  return {
+    size: limit,
+    timestamp: event['@timestamp'],
+    eventID: extractEventID(event),
+  };
+}
+
+export function paginate(pagination: PaginationParams, field: string, query: JsonObject) {
+  const { size, timestamp, eventID } = pagination;
+  query.sort = [{ '@timestamp': 'asc' }, { [field]: 'asc' }];
+  query.aggs = { total: { value_count: { field } } };
+  query.size = size;
+  if (timestamp && eventID) {
+    query.search_after = ([timestamp, eventID] as unknown) as JsonObject[];
```

Comment:
@oatkiller @peluja1012 the linter was giving me problems here. I think `JsonObject` is basically the same thing as the `JSONish` type but neither of them liked the assignment to a tuple of [number, string]. Any ideas?